modules/core/src/main/java/org/apache/ignite/internal/processors/query/aware/IndexRebuildState.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.aware;
+
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ * State of rebuilding indexes for the cache.
+ */
+public class IndexRebuildState {
```

Comment:
do you really need such class ? If you have cacheName - you have all you need, as i understand, the only thing you need to store is **boolean** flag, consisting complete info isn`t it ?

Reply:
At first I thought I could get rid of this class. But then I thought that there might be problems due to the fact that we cannot get the context of the cache in order to understand the persistent cache or not. And this is important in order to delete the **IndexRebuildCacheInfo** from the MetaStorage later.